 Don't go busy loop when waiting child process.

Before this change, code works like this: after gets POLLHUP from poll, it just
go a busy loop to do poll -> waitpid -> poll -> waitpid ... until child process
really exit. This doesn't make sense and is waste of CPU.

diff --git a/logwrapper/logwrap.c b/logwrapper/logwrap.c
index 030d2b8..13fc5fa 100644
--- a/logwrapper/logwrap.c
+++ b/logwrapper/logwrap.c
@@ -400,7 +400,7 @@ static int parent(const char *tag, int parent_read, pid_t pid,
         if (poll_fds[0].revents & POLLHUP) {
             int ret;
 
-            ret = waitpid(pid, &status, WNOHANG);
+            ret = TEMP_FAILURE_RETRY(waitpid(pid, &status, 0));
             if (ret < 0) {
                 rc = errno;
                 ALOG(LOG_ERROR, "logwrap", "waitpid failed with %s\n", strerror(errno));
